tests/rest/test_well_known.py

```diff
@@ -59,6 +59,30 @@ def test_client_well_known_no_public_baseurl(self) -> None:
 
         self.assertEqual(channel.code, HTTPStatus.NOT_FOUND)
 
+    @unittest.override_config(
+        {
+            "default_identity_server": "https://testis",
+        }
+    )
+    def test_client_well_known_only_default_identity_server(self) -> None:
+        channel = self.make_request(
+            "GET", "/.well-known/matrix/client", shorthand=False
+        )
+
+        self.assertEqual(channel.code, HTTPStatus.NOT_FOUND)
+
+    @unittest.override_config(
+        {
+            "extra_well_known_content": {"custom": False},
+        }
+    )
+    def test_client_well_known_only_extra_well_known(self) -> None:
+        channel = self.make_request(
+            "GET", "/.well-known/matrix/client", shorthand=False
+        )
+
+        self.assertEqual(channel.code, HTTPStatus.NOT_FOUND)
+
```

Comment:
I'm surprised these tests pass as written.

- override_config makes additions to the `default_config`; fields that aren't mentioned don't get changed
- a public_baseurl will be set based on the server_name from `default_config` (see also - https://github.com/matrix-org/synapse/pull/13035#issuecomment-1156441548)

Reply:
Ahhh. We only actually serve well-known if a public_baseurl is explicitly set:

https://github.com/matrix-org/synapse/blob/e87b7b329a48381b4a0c5309e98cd2548bbb0ad3/synapse/config/server.py#L273-L289